### DISCUSSION TITLE
Can't find MPI

### DISCUSSION THREAD

**Discussion by TangLaoya** [scivision/mumps-superbuild#86] [Q&A]
Dear author,

Thank you very much for providing so powerful cmake file. I am trying to use it to generate visual studio 2022 project, I have installed the latest Intel oneapi 2024.1 and running the cmake, the following error displayed:

`
Could NOT find MPI_C (missing: MPI_C_LIB_NAMES MPI_C_HEADER_DIR MPI_C_WORKS) 
Could NOT find MPI_Fortran (missing: MPI_Fortran_LIB_NAMES MPI_Fortran_F77_HEADER_DIR MPI_Fortran_MODULE_DIR MPI_Fortran_WORKS) 
CMake Error at C:/Program Files/CMake/share/cmake-3.29/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
  Could NOT find MPI (missing: MPI_C_FOUND MPI_Fortran_FOUND C Fortran)
Call Stack (most recent call first):
  C:/Program Files/CMake/share/cmake-3.29/Modules/FindPackageHandleStandardArgs.cmake:600 (_FPHSA_FAILURE_MESSAGE)
  C:/Program Files/CMake/share/cmake-3.29/Modules/FindMPI.cmake:1837 (find_package_handle_standard_args)
  CMakeLists.txt:45 (find_package)

`
I have set the environment both 
`I_MPI_ROOT` and `MPI_ROOT`
to
`D:\Program Files (x86)\Intel\oneAPI\mpi\latest\`

could you please help me to take a look at it?

Thanks,
Tang Laoya

**Comment by TangLaoya** [comment]
Dear scivision,

This modification works for me. I juast added the following code at line 26:

`set(ENV{Path} "$ENV{Path};$ENV{I_MPI_ROOT}/bin")`


Thanks,
Tang Laoya

[CMakeLists.txt](https://github.com/scivision/mumps/files/15242630/CMakeLists.txt)

**Comment by scivision** [comment]
Hi Tang, [FindMPI.cmake looks at `ENV{I_MPI_ROOT}` by default](https://cmake.org/cmake/help/latest/module/FindMPI.html#variables-for-locating-mpi), so I'm not sure why this change was needed on your system.
